source/mqtt_agent.c

```diff
@@ -499,7 +516,9 @@ static MQTTStatus_t processCommand( MQTTAgentContext_t * pMqttAgentContext,
 
     operationStatus = commandFunction( pMqttAgentContext, pCommandArgs, &commandOutParams );
 
-    if( ( operationStatus == MQTTSuccess ) && commandOutParams.addAcknowledgment )
+    if( ( operationStatus == MQTTSuccess ) &&
+        commandOutParams.addAcknowledgment &&
```

Comment:
`addAcknowledgment` sounds like a `bool`, so I would make this conditional more explicit:

```c
( commandOutParams.addAcknowledgement == true )
```

Reply:
That was a deliberate choice since we haven't been using a lot of `== true` so far in this code. It's also not necessary for MISRA since coreMQTT includes stdbool.h. Does adding `== true` add any value here?

Reply:
I'm fine with this as-is as long as it passes all static checks.  I think it's just more readable if all of the expressions in the conditional have parentheses+explicit values to compare with.